cmd/sqlconfig/sqlconfig.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package sqlconfig
+
+type EndpointDetails struct {
+	Address string `mapstructure:"address"`
+	Port    int    `mapstructure:"port"`
+}
+
+type DockerDetails struct {
+	ContainerId string `mapstructure:"containerId"`
+	Image       string `mapstructure:"image"`
+}
+
+type Endpoint struct {
+	DockerDetails   `mapstructure:"docker" yaml:"docker"`
+	EndpointDetails `mapstructure:"endpoint" yaml:"endpoint"`
+	Name            string `mapstructure:"name"`
+}
+
+type ContextDetails struct {
+	Endpoint string `mapstructure:"endpoint"`
+	User     string `mapstructure:"user"`
+}
+
+type Context struct {
+	ContextDetails `mapstructure:"context" yaml:"context"`
+	Name           string `mapstructure:"name"`
+}
+
+type UserDetails struct {
```

Comment:
>pe UserDetails s [](http://example.com/codeflow?start=2&length=16)

this seems "basic auth" specific
Perhaps User needs an authentication type value and a map that holds parameters specific to the auth type.

Where do we hook up the -U switch on the CLI as an override of the user name of the current context?